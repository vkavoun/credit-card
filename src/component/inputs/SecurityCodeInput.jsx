import React from 'react';
import PropTypes from 'prop-types';

export default function SecurityCodeInput({ securityCode, handleSecurityCodeInput }) {
  return (
    <div>
      <label>CVV (back of card)*</label>
      <input
        id="card-security-code-manual-entry__keyed-entry-input"
        name="card-security-code-manual-entry__keyed-entry-input"
        data-testid="security-code"
        type="tel"
        maxLength={3}
        value={securityCode}
        onChange={e => handleSecurityCodeInput(e)}
      />
    </div>
  );
}

SecurityCodeInput.propTypes = {
  securityCode: PropTypes.string.isRequired,
  handleSecurityCodeInput: PropTypes.func.isRequired,
};
