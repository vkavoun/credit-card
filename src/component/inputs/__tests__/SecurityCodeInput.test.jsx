import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import SecurityCodeInput from '../SecurityCodeInput';

describe('SecurityCodeInput', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <SecurityCodeInput securityCode={''} handleSecurityCodeInput={jest.fn()} />,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<SecurityCodeInput securityCode={''} handleSecurityCodeInput={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
