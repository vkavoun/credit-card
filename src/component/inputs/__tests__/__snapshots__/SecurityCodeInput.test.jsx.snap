// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityCodeInput renders correctly 1`] = `
<div>
  <label>
    CVV (back of card)*
  </label>
  <input
    data-testid="security-code"
    id="card-security-code-manual-entry__keyed-entry-input"
    maxLength={3}
    name="card-security-code-manual-entry__keyed-entry-input"
    onChange={[Function]}
    type="tel"
    value=""
  />
</div>
`;
